src/Tools/dotnet-stack/ReportCommand.cs

```diff
@@ -33,18 +33,37 @@ internal static class ReportCommandHandler
         /// <param name="console"></param>
         /// <param name="processId">The process to report the stack from.</param>
         /// <param name="name">The name of process to report the stack from.</param>
-        /// <param name="output">The output path for the collected trace data.</param>
+        /// <param name="diagnosticPort">Path to the diagnostic port to be used.</param>
         /// <param name="duration">The duration of to trace the target for. </param>
         /// <returns></returns>
-        private static async Task<int> Report(CancellationToken ct, IConsole console, int processId, string name, TimeSpan duration)
+        private static async Task<int> Report(CancellationToken ct, IConsole console, int processId, string name, string diagnosticPort, TimeSpan duration)
         {
             string tempNetTraceFilename = Path.GetRandomFileName() + ".nettrace";
             string tempEtlxFilename = "";
+            DiagnosticsClient client = null;
 
             try
             {
-                // Either processName or processId has to be specified.
-                if (!string.IsNullOrEmpty(name))
+                IpcEndpointConfig portConfig = IpcEndpointConfig.Parse(diagnosticPort);
+
+                // Either processName, processId or diagnosticPort has to be specified.
+                if (!string.IsNullOrEmpty(portConfig.Address))
+                {
+                    if (processId != 0 || !string.IsNullOrEmpty(name))
+                    {
+                        Console.WriteLine("Can only specify either --diagnostic-port or --name or --process-id option");
+                        return -1;
+                    }
+
+                    if (!portConfig.IsConnectConfig)
+                    {
+                        Console.WriteLine("--diagnostic-port only supports connect mode.");
```

Comment:
Is this limitation to reduce the immediate implementation cost or for something else? I worry it will confuse users if our tools aren't handling this diagnostic-port argument uniformly. If we wanted to save implementation cost I'd propose not supporting it at all for now.

Reply:
Yes, dotnet-stack didn't support --diagnostic-port and I tried to reduce all changes needed and would still be nice to get dotnet-stack working with dsrouter in server-server, server-client mode. If we don't support it we won't be able to use dotnet-stack with dsrouter, so I believed it was better to at least add something than nothing.

Reply:
My personal preference would be to support it fully if it is important or leave it unsupported for now if not. However if you think it is important to get the partial solution quickly then I'd suggest we should at least change the error message to make it clear that it is dotnet-stack specifically that doesn't support this, not a general limitation in --diagnostic-port across our tools. Perhaps:
```
Console.WriteLine("dotnet-stack support for a listening mode diagnostic-port is not yet available");
```

Aside: If adding listen support to dotnet-stack is hard I think it means we need to improve how we've factored code for the tools. Ideally the code that supports all these different connection options should be somewhere that we share it easily.

Reply:
The logic is a little spread out, and each piece I touch tends to call for more refactoring, so a little afraid that this PR will just continue to bloat :), but I can of course take a look and see (don't think its that much), but will be some duplication coming in from dotnet-trace or dotnet-counters. The other tools have the whole process execution feature connected to the --diagnostic port as well, but maybe we could at least leave that outside for now and focus only on --diagnostic-port argument?

Reply:
> so a little afraid that this PR will just continue to bloat :)

Two options if you want to press the merge button sooner is to leave the functionality out for a later PR or only change the error message. While I error message only isn't my preference, I don't begrudge you if that is what you decide is the best course : )

Reply:
After discussing and thinking about it a little more, maybe we should just leave it out in this PR and then roll a new PR implementing support for --diagnostic-port in dotnet-stack.

Reply:
Removed it from this PR.